ui/org.eclipse.pde.ui/src/org/eclipse/pde/internal/ui/launcher/TracingBlock.java

```diff
@@ -58,6 +64,62 @@
 
 public class TracingBlock {
 
+	/**
+	 * This class encapsulates all calls to the tracing options manager and runs them using a
+	 */
+	private static final class TracingOptionsManagerDelegate {
+
+		private IRunnableContext fRunnableContext;
+
+		public void setRunnableContext(IRunnableContext runnableContext) {
+			fRunnableContext = runnableContext;
+		}
```

Comment:
If you prefer to have those methods bundled in a separate class (you could also just add a `IRunnableContext fRunnableContext` field to the `TracingBlock` class and have the methods there), I would make it a record and create a new record object whenever the context is reset (which I assume happens rarely).
```suggestion
	private static record TracingOptionsManagerDelegate(IRunnableContext runnableContext) {
```

Reply:
I made it a record, thank you for the tip!